lib/timeutils.c

```diff
@@ -448,10 +448,11 @@ int get_gmtoff(const struct tm *tp)
 #endif
 }
 
-static int format_iso_time(const struct tm *tm, suseconds_t usec, int flags, char *buf, size_t bufsz)
+static int format_iso_time(const struct tm *tm, long nsec, int flags, char *buf, size_t bufsz)
```

Comment:
Internally it always casts it to int64_t; would it be better to use "int64_t" rather than "long"? And do we need it signed rather than unsigned?

Reply:
`int32_t` would fit the definition from timespec(3type).

```
       tv_nsec is of an implementation-defined *signed* type capable of holding
       the specified range. [..]
       It can be safely down-cast to any concrete *32-bit* integer type
       for processing.
```
(emphasis mine)